mat64/dense_arithmetic.go

```diff
@@ -674,9 +620,10 @@ func (m *Dense) Exp(a Matrix) {
 			Rows:   r,
```

Comment:
It's a special case because Exp requires the matrix to be eye. I could rewrite it to use this code, but I would rather not.